joplin/base/views/joplin_search_views.py

```diff
@@ -0,0 +1,129 @@
+from django.shortcuts import get_object_or_404, redirect, render
+from django.http import HttpResponse
+from django.http.request import QueryDict
+from django.contrib.contenttypes.models import ContentType
+from django.core.paginator import Paginator
+from wagtail.core.models import Page, UserPagePermissionsProxy
+from wagtail.admin.views import pages
+from wagtail.admin import messages
+from wagtail.search.query import MATCH_ALL
+from wagtail.admin.forms.search import SearchForm
+from django.utils.translation import ugettext as _
+from django.urls import reverse
+from django.conf import settings
+from base.models import ServicePage, ProcessPage, InformationPage, TopicPage, TopicCollectionPage, DepartmentPage, Theme, OfficialDocumentPage, GuidePage
+from base.models.site_settings import JanisBranchSettings
+import json
+
+
+def search(request):
+    pages = all_pages = Page.objects.all().prefetch_related('content_type').specific()
+    q = MATCH_ALL
+    content_types = []
+    pagination_query_params = QueryDict({}, mutable=True)
+    ordering = None
+
+    if 'ordering' in request.GET:
+        if request.GET['ordering'] in ['content_type', '-content_type', 'owner', '-owner', 'title', '-title', 'latest_revision_created_at', '-latest_revision_created_at', 'live', '-live']:
+            ordering = request.GET['ordering']
+
+            # The content_type order column was added so Joplin search results could order alphabetically by content_type
+            if ordering == 'content_type':
+                pages = pages.order_by('content_type')
+            elif ordering == '-content_type':
+                pages = pages.order_by('-content_type')
+
+            # The owner order column was added so Joplin search results could order alphabetically by owner
+            if ordering == 'owner':
+                pages = pages.order_by('owner')
+            elif ordering == '-owner':
+                pages = pages.order_by('-owner')
```

Comment:
The `content_type` and `owner` field weren't being ordered when clicking the column header. To add the functionality, their names needed to be added to this condition here.